viewing_party/party.py

```diff
@@ -1,23 +1,161 @@
+    for genre, count in genre_count.items(): 
+        if count > x:
+            x = count
+            best_genre = genre
+    return best_genre
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def create_film_diff_sets(user_data, is_user):
+    user_set = set()
+    friend_set = set()
+
+    for film in user_data["watched"]:
+        user_set.add(film["title"])
+    for friend in user_data["friends"]:
+        for film in friend["watched"]:
+            friend_set.add(film["title"])
+
+    if is_user == True:
+        user_set.difference_update(friend_set)
+        return user_set
+    elif is_user == False:
+        friend_set.difference_update(user_set)
+        return friend_set
+
+
+
+def get_unique_watched(user_data):
+    user_set = create_film_diff_sets(user_data, True)
+    unique_films = []
+
+    for title in user_set:
+        for film in user_data[ "watched"]:
+            if title == film["title"]:
+                unique_films.append(film)
+    return unique_films
+
+
+def get_friends_unique_watched(user_data):
+    friend_set = create_film_diff_sets(user_data, False)
+    unique_films = []
+    duplicate = False
+
+    for title in friend_set:
+        for friend in user_data["friends"]:
+            for film in friend[ "watched"]:
+                 
+                 if title == film["title"]:
+                    unique_films.append(film)
+                    duplicate = True 
+                    break
+            
+            if duplicate == True:
+                duplicate = False
+                break        
+                    
+                
+            
+    return unique_films
+   
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    avialable_recs = []
+    friends_unique_watched = get_friends_unique_watched(user_data)
+    
+    for film in friends_unique_watched:
+        if film["host"] in user_data["subscriptions"]:
+            avialable_recs.append(film)
+
+    return avialable_recs
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    most_watched_genre = get_most_watched_genre(user_data)
+    unwatched = get_friends_unique_watched(user_data)
+    recomendations = []
+
+    for films in unwatched:
+        if films["genre"] == most_watched_genre:
+            recomendations.append(films)
+    return recomendations
+def get_rec_from_favorites(user_data):
+    unique_watched = get_unique_watched(user_data)
+    recomendations = []
+
+    for film in unique_watched:
+        for movie in user_data["favorites"]:
+            if movie["title"] == film["title"]:
+                recomendations.append(movie)
+
+    return recomendations

```

Comment:
Really nice functions! Tiny nitpicks:
- Make sure to add an empty line between functions.
- Try to double-check your spelling when possible. "Recommendations" has 2 'm's.